src/Http/Cookie/SetCookieMiddleware.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Http\Cookie;
+
+use Tempest\Http\HttpMiddleware;
+use Tempest\Http\Request;
+use Tempest\Http\Response;
+
+final readonly class SetCookieMiddleware implements HttpMiddleware
+{
+    public function __construct()
+    {
+    }
+
+    public function __invoke(Request $request, callable $next): Response
+    {
+        /** @var Response $response */
+        $response = $next($request);
+
+        foreach ($response->cookies()->all() as $cookie) {
+            $response->header('set-cookie', (string) $cookie);
```

Comment:
I wonder if there are any edge cases we need to be aware about using this over `setcookie`?